data_steward/cdr_cleaner/cleaning_rules/clean_survey_conduct_recurring_surveys.py

```diff
@@ -0,0 +1,176 @@
+
+There are multiple cleaning rules that clean the survey conduct table.
+This CR will clean survey_concept_id and survey_source_value for the surveys with custom concept ids.
+Another cleans survey_source_concept_id for all valid records.
+Then observations that do not have a valid survey will be dropped in another CR.
+
+Original Issues: DC-3082
+"""
+# Python imports
+import logging
+
+# Project imports
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule, query_spec_list
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from common import JINJA_ENV, SURVEY_CONDUCT, COPE_SURVEY_MAP
+
+LOGGER = logging.getLogger(__name__)
+
+JIRA_ISSUE_NUMBERS = ['DC-3082']
+
+DOMAIN_TABLES = [SURVEY_CONDUCT]
+REFERENCE_TABLES = [COPE_SURVEY_MAP]
+
+SANDBOX_SURVEY_CONDUCT = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_dataset_id}}.{{sandbox_table_id}}` AS (
+SELECT * FROM `{{project_id}}.{{dataset_id}}.survey_conduct`
+WHERE survey_conduct_id IN (SELECT questionnaire_response_id FROM `{{project_id}}.{{dataset_id}}.{{cope_survey_map}}`)
+AND (survey_concept_id NOT IN (2100000006, 2100000007, 2100000002, 2100000005, 2100000004, 2100000003, 905047, 905055, 765936, 1741006)
+     OR survey_source_value NOT IN ('AoUDRC_SurveyVersion_CopeDecember2020', 'AoUDRC_SurveyVersion_CopeFebruary2021', 'AoUDRC_SurveyVersion_CopeMay2020', 'AoUDRC_SurveyVersion_CopeNovember2020', 'AoUDRC_SurveyVersion_CopeJuly2020', 'AoUDRC_SurveyVersion_CopeJune2020','cope_vaccine1', 'cope_vaccine2', 'cope_vaccine3', 'cope_vaccine4')
+     )
+)
+""")
+
+CLEAN_SURVEY_CONDUCT = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{dataset_id}}.survey_conduct` sc
+SET
+survey_concept_id = CASE
+    WHEN cope_month = 'dec' THEN 2100000006
+    WHEN cope_month = 'feb' THEN 2100000007
+    WHEN cope_month = 'may' THEN 2100000002
+    WHEN cope_month = 'nov' THEN 2100000005
+    WHEN cope_month = 'july' THEN 2100000004
+    WHEN cope_month = 'june' THEN 2100000003
+    WHEN cope_month = 'vaccine1' THEN 905047
+    WHEN cope_month = 'vaccine2' THEN 905055
+    WHEN cope_month = 'vaccine3' THEN 765936
+    WHEN cope_month = 'vaccine4' THEN 1741006
+    ELSE sc.survey_concept_id
+    END,
+survey_source_value = CASE
+    WHEN cope_month = 'dec' THEN 'AoUDRC_SurveyVersion_CopeDecember2020'
+    WHEN cope_month = 'feb' THEN 'AoUDRC_SurveyVersion_CopeFebruary2021'
+    WHEN cope_month = 'may' THEN 'AoUDRC_SurveyVersion_CopeMay2020'
+    WHEN cope_month = 'nov' THEN 'AoUDRC_SurveyVersion_CopeNovember2020'
+    WHEN cope_month = 'july' THEN 'AoUDRC_SurveyVersion_CopeJuly2020'
+    WHEN cope_month = 'june' THEN 'AoUDRC_SurveyVersion_CopeJune2020'
+    WHEN cope_month = 'vaccine1' THEN 'cope_vaccine1'
+    WHEN cope_month = 'vaccine2' THEN 'cope_vaccine2'
+    WHEN cope_month = 'vaccine3' THEN 'cope_vaccine3'
+    WHEN cope_month = 'vaccine4' THEN 'cope_vaccine4'
+    ELSE sc.survey_source_value
+    END
+FROM (SELECT *
```

Comment:
nit: You can simplify this SQL using `JOIN` instead of `LEFT JOIN`. Something like this:
```
FROM (SELECT *
      FROM `{{project_id}}.{{dataset_id}}.survey_conduct` sc
      JOIN `{{project_id}}.{{dataset_id}}.{{cope_survey_map}}` m
      ON m.questionnaire_response_id = sc.survey_conduct_id
      ) sub
```